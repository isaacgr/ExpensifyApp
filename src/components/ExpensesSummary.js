import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import total from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00')
    return (
      <div>
        {
          expenseCount === 0 ? '' :
          <p>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</p>
        }
      </div>
    )
}

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
  console.log(visibleExpenses);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: total(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
