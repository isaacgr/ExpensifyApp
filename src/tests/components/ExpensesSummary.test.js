import React from 'react';
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary for one expense', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={1}
      expensesTotal={100}
     />
   )
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary for multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={3}
      expensesTotal={500}
     />
   )
  expect(wrapper).toMatchSnapshot()})
