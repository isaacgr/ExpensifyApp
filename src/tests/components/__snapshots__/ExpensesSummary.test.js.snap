// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $5.00
  </p>
</div>
`;

exports[`should render ExpensesSummary for one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </p>
</div>
`;
